src/lattice.cpp

```diff
@@ -710,69 +711,84 @@ const
 std::array<int, 3>
 HexLattice::get_indices(Position r, Direction u) const
 {
-  // The implementation for HexLattice currently doesn't use direction
-  // information. As a result, we move the position slightly forward to
-  // determine what lattice index the particle is most likely to be in.
-  r += TINY_BIT * u;
-
   // Offset the xyz by the lattice center.
   Position r_o {r.x - center_.x, r.y - center_.y, r.z};
   if (is_3d_) {r_o.z -= center_.z;}
 
-  // Index the z direction.
-  std::array<int, 3> out;
-  if (is_3d_) {
-    out[2] = std::floor(r_o.z / pitch_[1] + 0.5 * n_axial_);
-  } else {
-    out[2] = 0;
-  }
-
   // Convert coordinates into skewed bases.  The (x, alpha) basis is used to
   // find the index of the global coordinates to within 4 cells.
   double alpha = r_o.y - r_o.x / std::sqrt(3.0);
-  out[0] = std::floor(r_o.x / (0.5*std::sqrt(3.0) * pitch_[0]));
-  out[1] = std::floor(alpha / pitch_[0]);
+  int ix = std::floor(r_o.x / (0.5*std::sqrt(3.0) * pitch_[0]));
+  int ia = std::floor(alpha / pitch_[0]);
 
   // Add offset to indices (the center cell is (i_x, i_alpha) = (0, 0) but
   // the array is offset so that the indices never go below 0).
-  out[0] += n_rings_-1;
-  out[1] += n_rings_-1;
+  ix += n_rings_-1;
+  ia += n_rings_-1;
+
+  // Index the z direction, accounting for coincidence
+  int iz{};
```

Comment:
"Explicit is better than implicit." - Zen of Python

Use `int iz = 0` or at least `int iz {0}`